docs/DeveloperGuide.md

```diff
@@ -184,7 +184,57 @@ A `TableNumber` object stores a table number as an integer. It is wrapped in an
   * Pros: Allows for greater flexibility in implementing features that are specific to either guests or vendors.
   * Cons: More code duplication.
 
+### Filter Guests/ Vendors Feature
 
+The implementation of the `filter` command allows the user to view a filtered list for both guests and vendors.
+The filtering is based on an AND search, for example, `guest filter n/John r/yes` will show only guests that have "John" in their 
+name and have also agreed to come to the wedding.
+The values in the parameter have to exactly match the keywords. for example `guest filter n/John` will return a guest by the
+name of `John Doe`, however a guest with the name `Johnathan` will not be returned.
+
+#### Proposed Implementation
+The filtering logic is done with predicate classes that implement Java's Predicate interface.
+<puml src="diagrams/FilterPersonPredicateClassDiagram.puml" alt="FilterPersonPredicateClassDiagram" />
+<puml src="diagrams/FilterGuestPredicateClassDiagram.puml" alt="FilterGuestPredicateClassDiagram" />
+
+The following sequence diagrams shows how the `filter` command works.
+<puml src="diagrams/FilterGuestSequenceDiagram.puml" alt="FilterGuestSequenceDiagram" />
+<puml src="diagrams/FilterGuestSequenceDiagramRef.puml" alt="FilterGuestSequenceDiagramRef" />
+<puml src="diagrams/FilterVendorSequenceDiagram.puml" alt="FilterVendorSequenceDiagram" />
+<puml src="diagrams/FilterVendorSequenceDiagramRef.puml" alt="FilterVendorSequenceDiagramRef" />
```

Comment:
Is it necessary to provide repeated diagrams for guest and vendor that are so similar?